src/WebJobs.Script/Config/ScriptJobHostOptionsSetup.cs

```diff
@@ -51,6 +51,12 @@ public void Configure(ScriptJobHostOptions options)
                 {
                     options.FileLoggingMode = fileLoggingMode.Value;
                 }
+
+                var sendCanceledInvocationsToTheWorker = jobHostSection.GetSection("SendCanceledInvocationsToTheWorker");
+                _applicationHostOptions.Value.SendCanceledInvocationsToTheWorker = sendCanceledInvocationsToTheWorker.Value is null
```

Comment:
Cross configuration of different options objects from another configuration is not recommended. This is not deterministic and introduces race conditions. Options are lazily constructed on first import, so this only works if both `ScriptJobHostOptions` and `ScriptApplicationHostOptions` are imported. This is not intuitive and can easily break in the future.

We will need to figure out a deterministic way to get this setting where we need it.

Reply:
Yup agree, planning to sync with @fabiocav on this soon to see where is the appropriate place for this config to be

Reply:
I had a sync with Fabio and looks like we can use this the host manager to get the jobhost options and avoid using the webhost options completly:

`_serviceProvider.GetService(typeof(IOptions<ScriptJobHostOptions>));`